### PR TITLE
Updating ethcore and ethjson crates to Rust Edition 2018

### PR DESCRIPTION
**Pull request by karim-agha**
This change updates the rust edition for ethcore and ethjson to Rust 2018.

### CODE REVIEW

File: crates/ethcore/res/chainspec/test/yolo3_test.json [hunk 1]
```diff
@@ -0,0 +1,204 @@
+{
```
Comment (adria0) [inline crates/ethcore/res/chainspec/test/yolo3_test.json:1]: IMO this should be added in another PR

Comment (karim-agha) [reply crates/ethcore/res/chainspec/test/yolo3_test.json:1]: Agreed.

**Review by jomuel — APPROVED** [review]

File: crates/ethjson/src/state/log.rs [hunk 2]
```diff
@@ -50,3 +53,13 @@ mod tests {
         // TODO: validate all fields
     }
 }
+
+impl From<Log> for LogEntry {
```
Comment (rakita) [inline crates/ethjson/src/state/log.rs:57]: Is this even used, do you need it else somewhere else?

File: crates/ethjson/src/transaction/transaction.rs [hunk 3]
```diff
@@ -45,10 +47,37 @@ pub struct Transaction {
     pub v: Uint,
 }
 
+impl From<Transaction> for UnverifiedTransaction {
```
Comment (rakita) [inline crates/ethjson/src/transaction/transaction.rs:50]: I think this is removed from code because it is unused. Do you have usage for it?

**Review by rakita — APPROVED** [review]
lgtm. Just you can remove two added `From` `Transaction` and `Log` and can add them in pull request when you need them. name of pull request could be without `update ethcore/ethjson`. And btw great job I wanted to do this for months :)

File: crates/ethjson/src/transaction/transaction.rs (hunk 3 above)
Comment (karim-agha) [reply crates/ethjson/src/transaction/transaction.rs:50]: Afaik it’s used in tests. It’s also part of removing the dependency of ethcore on ethjson. I’m having lunch now and I’ll double check it ASAP when I get home.

Comment (rakita) [reply crates/ethjson/src/transaction/transaction.rs:50]: It was modified by dusan and removed when test started supporting AccessList, it is deadcode now.

Comment (karim-agha) [reply crates/ethjson/src/transaction/transaction.rs:50]: Sure, I'll remove it in a follow-up PR.
